### PR TITLE
feature/gh-1394-enable-sorting-of-export-data

### PR DESCRIPTION
**Pull request by popovicbojan**
Resolves: #1394

### CODE REVIEW

File: src/main/java/eu/execom/hawaii/service/DataExportService.java [hunk 1]
```diff
@@ -122,22 +133,92 @@ private String assignDateRange(LocalDate startDate, LocalDate endDate) {
...
+    String defaultProperty = DEFAULT_SORT_PROPERTIES.get(exportType);
+    return Sort.by(new Sort.Order(defaultDirection, defaultProperty));
+  }
+
+  /**
+   * Throws exception if number of sort properties exceeds maximum allowed.
+   *
+   * @param propertiesLength number of properties for multiple sorting.
+   */
+  private void throwExceptionIfNumberOfSortPropertiesExceeded(int propertiesLength) {
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/service/DataExportService.java:183]: Please rename this method to `validateNumberOfSortProperties`.

File: src/main/java/eu/execom/hawaii/service/DataExportService.java [hunk 2]
```diff
@@ -122,22 +133,92 @@ private String assignDateRange(LocalDate startDate, LocalDate endDate) {
...
+    if (propertiesLength > MAX_SORT_PROPERTIES) {
+      String message = String.format(EXCEEDED_NUMBER_OF_PROPERTIES_MASSAGE, MAX_SORT_PROPERTIES);
+      log.error(message);
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/service/DataExportService.java:186]: Exception should be logged so you can skip this line.

File: src/main/java/eu/execom/hawaii/service/DataExportService.java [hunk 3]
```diff
@@ -122,22 +133,92 @@ private String assignDateRange(LocalDate startDate, LocalDate endDate) {
...
+
+  /**
+   * Checks if sort direction is set for each property. If not default direction ASC will be set.
+   *
+   * @param propertiesLength number of sort properties.
+   * @param directions list of set directions.
+   */
+  private void checkIfSortDirectionIsSetForEveryProperty(int propertiesLength, List<Sort.Direction> directions) {
+    int directionsLength = directions.size();
+    if (propertiesLength > directionsLength) {
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/service/DataExportService.java:199]: Can't you skip the `if` check and just loop and add missing `Sort.Direction`s?

Comment (popovicbojan) [reply src/main/java/eu/execom/hawaii/service/DataExportService.java]: Yes, you are right, `if` check is unnecessary.

File: src/main/frontend/src/pages/data-export/DataExport.js [hunk 4]
```diff
@@ -18,16 +18,25 @@ import DrawerCalendarInput, {
...
   ALLOWANCES_REPORT,
-  LEAVE_PROFILES_REPORT
+  LEAVE_PROFILES_REPORT,
+  ASC,
+  DESC
 } from '@/helpers/enum';
 
 class DataExport extends Component {
+  state = {
+    directions: [DESC, ASC, ASC]
```
Comment (aberonja) [inline src/main/frontend/src/pages/data-export/DataExport.js:37]: Why do we have ASC twice?

Comment (popovicbojan) [reply src/main/frontend/src/pages/data-export/DataExport.js:37]: We are sorting data by three properties defined by the name of the property and sort direction. Initial sort direction values are `desc, asc, asc`.

**Pull request merged**